code/datums/perks/fate.dm

```diff
@@ -214,12 +214,19 @@
 	..()
 	if(holder)
 		holder.sanity.max_level += 20
+		if(!get_active_mutation(holder, MUTATION_OBORIN))
+			var/datum/mutation/M = new MUTATION_OBORIN
+			M.imprint(holder)
 		spawn(1)
-			holder.update_client_colour() //Handle the activation of the colourblindness on the mob.
+			holder?.update_client_colour() //Handle the activation of the colourblindness on the mob.
```

Comment:
why ?. , this should 100% runtime if the holder is actually null. because imaking applying perks to null..

Reply:
Perk is not applied to null holder, but holder may be gone when `update_client_colour()` is called.
It does not runtime with `?.`, but does without.

Reply:
why would the holder be gone ?

Reply:
When perk is added or removed, it calls for related mutation to be added/removed as well, and vice versa.
That causes perk addition/removal to being called twice (initial and from mutation).
Addition of `?.` is simplest way to prevent runtime. Could be replaced with extra checks, but dunno if that's necessary.

Reply:
Actually null holder is not the case when perk is added, only when removed.